RIOT-OS/RIOT
# tests/unittests: split out atomic_utils unittests

### Contribution description

#14331 broke the CI because unittests don't fit anymore on arm7 based boards (msba2 and mcb2388), this PR move the sys atomic_utils from the unittests to their own test application.




### Testing procedure

- A green Murdock
- Check that `tests/sys_atomic_utils_unittests` are working:

<details>

```
$ BUILD_IN_DOCKER=1 make -C tests/sys_atomic_utils_unittests all test --no-print-directory 
Launching build container using image "riot/riotbuild:latest".
docker run --rm --tty --user $(id -u) -v '/usr/share/zoneinfo/Europe/Paris:/etc/localtime:ro' -v '/work/riot/RIOT:/data/riotbuild/riotbase:delegated' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'        -w '/data/riotbuild/riotbase/tests/sys_atomic_utils_unittests/' 'riot/riotbuild:latest' make     all
Building application "tests_sys_atomic_utils_unittests" for "native" with MCU "native".

"make" -C /data/riotbuild/riotbase/boards/native
"make" -C /data/riotbuild/riotbase/boards/native/drivers
"make" -C /data/riotbuild/riotbase/core
"make" -C /data/riotbuild/riotbase/cpu/native
"make" -C /data/riotbuild/riotbase/cpu/native/periph
"make" -C /data/riotbuild/riotbase/cpu/native/stdio_native
"make" -C /data/riotbuild/riotbase/drivers
"make" -C /data/riotbuild/riotbase/drivers/periph_common
"make" -C /data/riotbuild/riotbase/sys
"make" -C /data/riotbuild/riotbase/sys/auto_init
"make" -C /data/riotbuild/riotbase/sys/embunit
"make" -C /data/riotbuild/riotbase/sys/luid
"make" -C /data/riotbuild/riotbase/sys/random
"make" -C /data/riotbuild/riotbase/sys/random/tinymt32
"make" -C /data/riotbuild/riotbase/sys/test_utils/interactive_sync
   text	   data	    bss	    dec	    hex	filename
  52602	    728	  47804	 101134	  18b0e	/data/riotbuild/riotbase/tests/sys_atomic_utils_unittests/bin/native/tests_sys_atomic_utils_unittests.elf
r
/work/riot/RIOT/tests/sys_atomic_utils_unittests/bin/native/tests_sys_atomic_utils_unittests.elf  
RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

Help: Press s to start test, r to print it is ready
READY
s
START
main(): This is RIOT! (Version: 2021.01-devel-832-gfc8b1-pr/tests/sys_atomic_unittest)
.......
OK (7 tests)

```

</details>




### Issues/PRs references

Restore the CI that was broken in #14331
